test/test_Gaussian/main.cpp

```diff
@@ -71,70 +71,70 @@ int main()
 
         Gaussian gaussian_3d(3);
 
-        if (!(gaussian_3d.mean.array() == 0).all())
+        if (!(gaussian_3d.mean().array() == 0).all())
         {
-            std::cerr << "Default initialization of Gaussian mean is not zero, is\n" << gaussian_3d.mean << std::endl;
+            std::cerr << "Default initialization of Gaussian mean is not zero, is\n" << gaussian_3d.mean() << std::endl;
             return EXIT_FAILURE;
         }
         else
-            std::cout << "Default initialization of Gaussian mean is all zero:\n" << gaussian_3d.mean << std::endl;
+            std::cout << "Default initialization of Gaussian mean is all zero:\n" << gaussian_3d.mean() << std::endl;
 
-        if (!(gaussian_3d.covariance.array() == 0).all())
+        if (!(gaussian_3d.covariance().array() == 0).all())
         {
-            std::cerr << "Default initialization of Gaussian covariance is not zero, is\n" << gaussian_3d.covariance << std::endl;
+            std::cerr << "Default initialization of Gaussian covariance is not zero, is\n" << gaussian_3d.covariance() << std::endl;
             return EXIT_FAILURE;
         }
         else
-            std::cout << "Default initialization of Gaussian covariance is all zero:\n" << gaussian_3d.covariance << std::endl;
+            std::cout << "Default initialization of Gaussian covariance is all zero:\n" << gaussian_3d.covariance() << std::endl;
 
-        if (!(gaussian_3d.weight == 1.0))
+        if (!(gaussian_3d.weight() == 1.0))
         {
-            std::cerr << "Default initialization of Gaussian weight is not 1.0, is\n" << gaussian_3d.weight << std::endl;
+            std::cerr << "Default initialization of Gaussian weight is not 1.0, is\n" << gaussian_3d.weight() << std::endl;
             return EXIT_FAILURE;
         }
         else
-            std::cout << "Default initialization of Gaussian covariance is 1.0: " << gaussian_3d.weight << std::endl;
+            std::cout << "Default initialization of Gaussian weight is 1.0: " << gaussian_3d.weight() << std::endl;
 
         std::cout << "done!\n" << std::endl;
 
 
         std::cout << "Assign values to the Gaussian with 3D components..." << std::endl;
 
-        gaussian_3d.mean << 1, 2, 3;
+        gaussian_3d.mean() << 1, 2, 3;
 
-        gaussian_3d.covariance << 11, 12, 13,
-                                  21, 22, 23,
-                                  31, 32, 33;
+        gaussian_3d.covariance() << 11, 12, 13,
+                                    21, 22, 23,
+                                    31, 32, 33;
 
-        gaussian_3d.weight = 0.5;
+        gaussian_3d.weight() = 0.5;
 
-        if (!((gaussian_3d.mean(0) == 1) &&
-              (gaussian_3d.mean(1) == 2) &&
-              (gaussian_3d.mean(2) == 3)))
+        if (!((gaussian_3d.mean()(0) == 1) &&
```

Comment:
I don't really like this syntax, but let's keep it like this for now.
I suggest to add element-wise accessor to `GaussianMixture` and `Gaussian`, for example.
```cpp
gaussian_3d(0) = 2;
```
implies the first element of the mean vector;
```cpp
gaussian_mixture_3d(3, 0) = 2;
```
implies the first element of the mean vector of the 3rd gaussian component of the mixture.
Same reasoning with `ParticleSet`/`Particle`.

I would also do the same for the covariances.

Reply:
Implemented accessors for `Gaussian` and `GaussianMixture` (both mean and covariance).

As regards `ParticleSet` we should decide how to disambiguate between element-wise access to state and mean.

Reply:
Now it should be ok.

I also had to re-implement `Gaussian::covariance()` since the new method `Gaussian::covariance(const std:size_t i, const std::size_t j)` hides `GaussianMixture::covariance()` which was ok also for a Gaussian since the matrix of covariances shrinks to a single covariance matrix.

As regards `ParticleSet`, the methods for mean/covariance are inherited from `GaussianMixture` while I implemented element-wise access to the i-th state.